ml/src/main/scala/frameless/ml/TypedEstimator.scala

```diff
@@ -0,0 +1,21 @@
+package frameless
+package ml
+
+import frameless.ops.SmartProject
+import org.apache.spark.ml.{Estimator, Model}
+
+/**
+  * A TypedEstimator `fit` method takes as input a TypedDataset containing `Inputs`and
+  * return an AppendTransformer with `Inputs` as inputs and `Outputs` as outputs
+  */
+abstract class TypedEstimator[Inputs, Outputs, M <: Model[M]] private[ml] {
+  val estimator: Estimator[M]
+
+  def fit[T](ds: TypedDataset[T])(implicit smartProject: SmartProject[T, Inputs]): AppendTransformer[Inputs, Outputs, M] = {
+    val inputDs = smartProject.apply(ds)
+    val model = estimator.fit(inputDs.dataset)
```

Comment:
Hmm, this is an effect. Spark usually runs a bunch of actions in the `fit` methods. The rest of the frameless APIs wrap these effects in a `SparkDelay.delay`. Perhaps this should too?

Reply:
You're right, I forgot about that! I'll make the change.